QC Rework Quest Changes

DylanTaylor1 opened the pull request:
**THIS IS HEAVILY DEPENDENT ON THE QC REWORK https://github.com/GTNewHorizons/GT5-Unofficial/pull/2849**

The previous QC quest and "Play it Again, Sam" were both really unhelpful in understanding the QC and did not provide any numbers, equations, or really any information about the machine. They really just told the player to "do this" without much explanation as to why.

These new quests closely resemble the wiki page and provide a significant amount of information in an organized and readable manner. The goal is to give the player enough of an understanding that they can begin working with the QC without having to go to the wiki right away.

![2024-08-24_10 42 16](https://github.com/user-attachments/assets/f02d9c0c-06f2-4671-b5ad-5c889ac5ea81)

![2024-08-24_10 40 52](https://github.com/user-attachments/assets/766c9c3b-c5a3-43f8-9928-79b687bc7ce8)

![2024-08-24_10 41 03](https://github.com/user-attachments/assets/25852933-4e95-415b-bce7-5654dc73672d)

![2024-08-24_10 41 08](https://github.com/user-attachments/assets/f2ae305c-ab45-49da-ab62-1e5d5e16d6e3)

Pxx500 commented:
Small nitpick
![image](https://github.com/user-attachments/assets/a601c0d8-b3eb-47bb-9a28-7f57c2c0597d)
changing this to "e.g. [insert one or two multies]" will futureproof the quest so it won't need updating when a new multi using computation gets added.

Generally putting numbers into quests makes them outdated when any changes happen, it would generally be better to say something like "[x type of item] is eligible to generate computation, look in NEI tooltips to find the numbers by searching [xyz]" (new nei supports tooltip search) and actually adding info about those numbers into the tooltip, this makes maintaining questbook easier (though I understand just shoving numbers into the quest is easier so I'm not actually opposed to this PR)

Other than that it looks good 👍

DylanTaylor1 commented:
Thanks! And there could certainly be a whole debate on where to put the numbers. Previously they were not available anywhere besides the code and the wiki page which worked fine especially if there was just ONE best setup that everyone talked about. 

Putting them in the tooltip of the circuits might be a little confusing/cluttered because this is a very niche use of the circuits. Putting them in a quest is not super future proof like you mentioned, but at least it's somewhere and easy to edit. It could go many different ways but I think so long as there is one way to find them in-game and one way to find them out-of-game (wiki) is good.

chochem commented:
I would say tables of numbers fit better on a wiki page or spreadsheet than a quest.

maybe a custom NEI page would be a really nice in-game possiblity. But such a table in a quest text is pretty awkward

chochem commented on config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/QuantumComputer-AAAAAAAAAAAAAAAAAAAH1w==.json:
```diff
@@ -35,7 +35,7 @@
       "autoClaim:1": 0,
       "isSilent:1": 0,
-      "desc:8": "One of the most incredibly complicated machines, this will serve as the main structure for your Computation generation for the Research Station. \n\nFirst, you\u0027ll need a few things: the Quantum Computer itself, The Uncertainty Resolver, some good circuits and the Computer Racks to hold them, and at least a UV energy hatch (maybe start with a UV 4A energy hatch). The QC will  ...
+      "desc:8": "The Quantum Computer (QC) is an incredibly complicated machine used to generate computation for the Research Station and a few other multiblocks. There are a few things you need to understand before using one.\n\n§lCONSTRUCTION§r. Much like an assembly line, the QC is built as a series of slices. The first and last slice are the only valid locations for hatches. The slices in bet ...
```
not suggesting a basic sample setup without need for parameters to get people started is a pretty bad change imo.

chochem replied:
yea having it in the other quest is fine.

chochem commented on config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/QuantumMechanics-AAAAAAAAAAAAAAAAAAAKiw==.json:
```diff
@@ -0,0 +1,72 @@
+      "lockedProgress:1": 0,
+      "autoClaim:1": 0,
+      "isSilent:1": 0,
+      "desc:8": "The QC has 1-24 computer racks depending on the length of the machine. Each rack has four total slots for either circuits (to generate heat and computation) or vents (to dissipate heat). Although conceptually similar to a nuclear reactor, the location of the components inside does not matter at all. Every rack is also completely independent from every other rack which means it is ...
```
2-24

chochem commented on config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/TheNumbersMasonW-zzIqlbfdQl6Rn8cVlbiQsA==.json:
```diff
@@ -0,0 +1,42 @@
+      "autoClaim:1": 0,
+      "isSilent:1": 0,
+      "desc:8": "The following is an all inclusive list of valid rack components and their properties. Anything not listed has no effect on computation or heat. If you are struggling with all the numbers and equations, you might want to check the wiki.\n\n[note]Circuit - Computation - Heat Limit - Heat Constant[/note]\n\nCrystalprocessor - 60 - 2000 - 56\nCrystalprocessor Assembly - 80 - 2000 - 5 ...
+    }
```
I dont think this quest is a good idea

DylanTaylor1 commented:
Alrighty no worries, I deleted the quest with all the rack component values and adjusted the example setup.

![2024-08-24_12 55 23](https://github.com/user-attachments/assets/bf205789-b9c9-4885-bec4-44595491f4fd)

chochem commented:
I am also happy to hear other opinions on how to best represent that data:
- custom nei page (similar to how fuel values are shown)
- wiki
- tooltip? maybe not great as its just one small use from the perspective of a circuit.
- a table in the QB?

but yea for me at least a table in the QB does not seem like a great option. QB can/should of course have a link to the wiki if the data is there.
